WordPress/Classes/Services/PostService.m

```diff
@@ -604,6 +604,8 @@ - (void)trashRemotePostWithPost:(AbstractPost*)post
     
     RemotePost *remotePost = [self remotePostWithPost:post];
     id<PostServiceRemote> remote = [self.postServiceRemoteFactory forBlog:post.blog];
+    post.statusAfterSync = PostStatusTrash;
+    post.latest.statusAfterSync = PostStatusTrash;
```

Comment:
I'm a little bit concerned about this change because we're changing `statusAfterSync` before the sync (which is the `trashPost` below).

Ideally, we would have another property on the post to identify that the post has been trashed locally — but I think pursuing that involves Core Data and some other changes that are out of scope for this task.

Plus, on `failureBlock` below we would have to handle resetting `statusAfterSync` in case it fails.